website/static/js/share/results.js

```diff
@@ -37,79 +38,155 @@ Results.controller = function(vm) {
     self.vm = vm;
     self.vm.resultsLoading = m.prop(false);
 
+    self.renderTitleBar = function(result) {
```

Comment:
though if this will be a lengthy refactor, it may be better to just leave it as it is, or make a second PR with the goal of refactoring, since Jeff has hinted a few times that he wants the SHARE page rewritten sometime soon-ish, and the refactor was not a primary goal of this PR.